pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,472 @@
+        cast=lambda x: int(float(x)),
+        validator=strict_discrete_set,
+        values=SCAN_POINT_VALUES,
+    )
+
+    id = Instrument.measurement(
+        "*IDN?",
+        """Get the identification of the instrument (list of strings)""",
+        cast=str,
+    )
+
+    sn = Instrument.measurement(
+        "OUTPSERN",
+        """Get the serial number for the instrument""",
+    )
+
+    options = Instrument.measurement(
+        "OUTPOPTS",
+        """Get the installed options for the instrument""",
+    )
+
+    @property
+    def manu(self):
+        """Get the manufacturer of the instrument."""
+        if self._manu == "":
+            self._manu, self._model, _, self._fw = self.id
+        return self._manu
+
+    @property
+    def model(self):
+        """Get the model of the instrument."""
+        if self._model == "":
+            self._manu, self._model, _, self._fw = self.id
+        return self._model
+
+    @property
+    def fw(self):
+        """Get the firmware of the instrument."""
+        if self._fw == "":
+            self._manu, self._model, _, self._fw = self.id
+        return self._fw
+
+    def set_fixed_frequency(self, frequency):
+        """Set the sweep to be a fixed frequency in Hz."""
+        self.start_frequency = frequency
+        self.stop_frequency = frequency
+        self.scan_points = 3
+
+    @property
+    def measuring_parameter(self):
+        """Get the active Scattering or Measuring Parameter being measured.
+        (str from ['S11', 'S12', 'S21', 'S22', 'A/R', 'B/R', 'A/B', 'A', 'B', 'R'])"""
+        for key in self.MEASURING_PARAMETER_MAP:
+            if int(self.ask(f"{self.MEASURING_PARAMETER_MAP[key]}?")) == 1:
+                return key
+        return None
+
+    @measuring_parameter.setter
+    def measuring_parameter(self, value):
+        """Set the active Measuring Parameter to be measured.
+        (str from ['S11', 'S12', 'S21', 'S22', 'A/R', 'B/R', 'A/B', 'A', 'B', 'R'])"""
+        if value in self.MEASURING_PARAMETER_MAP:
+            self.write(self.MEASURING_PARAMETER_MAP[value])
+        else:
+            raise ValueError(
+                f"Invalid value '{value}' scattering parameter requested for \
+                {self._manu} {self._model}. Valid values are: {self.MEASURING_PARAMETER_MAP}"
+            )
+
+    IFBW = Instrument.control(
+        "IFBW?",
+        "IFBW%d",
+        f"""Control the IF Bandwidth of the instrument for a scan sweep.
+        (int from the set {ALLOWED_BANDWIDTH}).""",
+        cast=lambda x: int(float(x)),
+        validator=strict_discrete_set,
+        values=ALLOWED_BANDWIDTH,
+    )
+
+    def reset(self):
+        """Reset the instrument. May cause RF Output power to be enabled!"""
+        self.write("*RST")
+        sleep(0.25)
+
+    def scan(self, timeout=10):
+        """Initiates a scan with the number of averages specified and
+        blocks until the operation is complete.
```

Comment:
Docstrings should be in imperative mode:
```suggestion
        """Initiate a scan with the number of averages specified and
        block until the operation is complete.
```

Reply:
Fixed